lib/shipcloud.rb

```diff
@@ -89,9 +93,10 @@ def self.api_key=(api_key)
   # @param [String] optional api_key The api key. If no api key is given, Shipcloud.api_key will
   # be used for the request
   # @return [Array] The parsed JSON response.
-  def self.request(http_method, api_url, data, api_key: nil)
+  def self.request(http_method, api_url, data, api_key: nil, affiliate_id: nil)
     api_key ||= Shipcloud.api_key
+    affiliate_id ||= Shipcloud.affiliate_id
     info = Request::Info.new(http_method, api_url, api_key, data)
-    Request::Base.new(info).perform
+    Request::Base.new(info, affiliate_id).perform
```

Comment:
What do you think about adding the `affiliate_id` to the `Request::Info` similar to the `api_key` and passing it on as part of the `Request::Info` instance? Or did you have a special reason for passing it on separately?

Reply:
Yes, I tried to add `affiliate_id` as a part of `Request::Info` instance, but there is a limitation to the number of arguments, which can be passed to a method. It cannot exceed 4.  `affiliate_id ` would have been the fifth one, so I switched to the current solution.

Reply:
Done